src/DataAPI.jl

```diff
@@ -343,6 +343,9 @@ $STYLE_INFO
 
 If `default` is passed then return it if reading metadata is supported but
 mapping for `key` is missing. If `style=true` return `(default, :default)`.
+
+If `key` is not passed return a dictionary mapping all metadata keys to
+metadata values associated with object `x`.
```

Comment:
I would document this as a separate method given that a lot of the text above doesn't apply when `key` isn't passed.

Reply:
done